demo/site/src/app/block-preview/[domain]/[language]/page/page.tsx

```diff
@@ -1,35 +1,36 @@
 "use client";
 
-import { useBlockPreviewFetch, useIFrameBridge } from "@comet/site-nextjs";
+import { createFetchInMemoryCache, useIFrameBridge } from "@comet/site-nextjs";
 import { type PageContentBlockData } from "@src/blocks.generated";
 import { PageContentBlock } from "@src/documents/pages/blocks/PageContentBlock";
 import { withBlockPreview } from "@src/util/blockPreview";
+import { createGraphQLFetch } from "@src/util/graphQLClient";
 import { recursivelyLoadBlockData } from "@src/util/recursivelyLoadBlockData";
 import { useEffect, useState } from "react";
 
+const cachingFetch = createFetchInMemoryCache(fetch);
+
 export default withBlockPreview(() => {
     const iFrameBridge = useIFrameBridge();
 
-    const { fetch, graphQLFetch } = useBlockPreviewFetch();
```

Comment:
not needed anymore, this gets much simpler. Possible because since we have persisted queries createGraphQLFetch does work on server AND client side.